### ISSUE TITLE
No "Farthest from" region when distance is not checked

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab-basics#206]
I hear "Closest to", but I do not hear "Farthest from" when exploring with Distance not checked.
I know there was a change to the overall range from 9.6 km to 10 km shortly before or after publication. I am curious if that accidentally affected the single-value region for "Farthest from"?

Or am I missing something?

Region (qualitative- distance) | Farthest from {{mass 2}} | Extremely from {{mass 2}} | Very far from {{mass 2}} | Far from {{mass 2}} | Not so close to {{mass 2}} | Close to {{mass 2}} | Very close to {{mass 2}} | Extremely close to {{mass 2}} | Closest to {{mass 2}}
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
New Regions Ranges | 9.6 | 9.5 - 8.6 (0.9 m) | 8.5 - 7.6 (0.9 m) | 7.5 - 5.6 (0.9 m) | 4.5 - 3.6 (0.9 m) | 3.5 - 2.6 (0.9 m) | 2.5 - 1.6 (0.9 m) | 1.5 - 0.7 (0.8 m) | 0.6

The above table I copied from the design doc. Do I need to recalculate the region values to allow for 10.0 km to be "Farthest from mass 1" or "Farthest from mass 2"?

**Comment by zepumph** [comment]
I think that recalculation may be required unfortunately. It would have been nice to have normalized these values into ranged percents from 0-1. Would you like me to convert them now that the max has increased to 10?

Please note that this is a problem with GFL as well. And it effects the recent MR I did, which is likely why we hear this issue in the published version.

**Comment by terracoda** [comment]
Oh yes, we should definitely be using percentages or whatever scales.

**Comment by terracoda** [comment]
@zepumph, for the record, I just tested GFL regular on master and I hear "farthest from".

**Comment by terracoda** [comment]
@zepumph, I am going to assign @jessegreenberg, too, since he is working on this sim now for self-voicing. Perhaps, @jessegreenberg can address this issue?

**Comment by zepumph** [comment]
Noting here that this is an issue in the published version, and it needs to be MR'd because the fix in https://github.com/phetsims/gravity-force-lab/issues/265 broken the description. Sorry I haven't gotten to this yet.

**Comment by terracoda** [comment]
No apologies necessary :-)

**Comment by terracoda** [comment]
I need to get this region description issue addressed for the Self-Voicing design of GFLB.

In recent interview a learner described the reference to "a track" a little strange, that it would be useful for the sim to just say "Farthest from mass 1."

I would like to see how the presence of this region affects possible other changes to the design.

I am going to mention @jessegreenberg so he is aware of this issue.

Maybe when self-voicing work starts on RaP would be a goo time to address this issue.

**Comment by terracoda** [comment]
Will these percentage ranges work for all ISLC sims?

From the Design Doc:
**Table: (1.3 - 10 km) Distance Region Names & Ranges - Described distance between centers of spheres**
- We still have 7 multi-value regions and 2 single-value regions (the edges)
- The maximum distance between centers = 10 km with or without Constant Size checked
- The minimum distance between centers varies between 1.3 km and 2.5 km depending on the chosen mass for the spheres, thus when a learner explores with larger masses and without constant size checked, the centers of the spheres cannot reach some of the smaller distance regions

![Screen Shot 2021-05-26 at 2 41 04 PM](https://user-images.githubusercontent.com/1902116/119702849-7b396a00-be30-11eb-8de5-301c6202692b.png)

**Comment by terracoda** [comment]
@jessegreenberg, I am marking this for Friday's meeting or forthe next meeting when we discuss GFLB Voicing design.

**Comment by jessegreenberg** [comment]
@terracoda is this something that still requires work?

**Comment by terracoda** [comment]
Yes, @jessegreenberg, this was a bug introduce with a maintenance release that changed the distance from 9.6 to 10 km. I think the issue was that the qualitative distance scale was hard coded.

It am hoping the precentages I worked out in https://github.com/phetsims/gravity-force-lab-basics/issues/206#issuecomment-848960764 will fix the issue. 

You might need to check with @zepumph about how or is this scale is shared across all the ISLC repos.

I don't think this issue is in the published version of GFL regular. Oh, I think that is becasue there is no distance arrow in that sim, so users do not swap between qualitative and quantitative distance. They just get quantitative distance with progress indicators and landmarks.

**Comment by jessegreenberg** [comment]
@zepumph and I discussed this issue today and we reviewed the issue and the proposal in https://github.com/phetsims/gravity-force-lab-basics/issues/206#issuecomment-848960764. I will work on updating the ranges to be normalized and use the percentages provided.

Once working I will assign to @terracoda and @zepumph to review. @zepumph might see through a maintenance release of GFL (regular) with this fix. GFL:B will be published soon and does not need to be MR'd any more.

**Comment by terracoda** [comment]
I am not sure this bug is in GFL regular because there is no distance checkbox in that sim, so no qualitative distance.

**Comment by zepumph** [comment]
Yes thanks, I see you are totally right. Although the bug is in GFL, and should be fixed in https://github.com/phetsims/gravity-force-lab/blob/5993821a779f528e92dbf2634dc8dba8b6a83aea/js/view/describers/GravityForceLabPositionDescriber.js#L39-L70

(or maybe deleted?)

. . . it seems like it isn't used, I see that the only PositionDescriber that switches `useQuantitativeDistance` to false is in GFLB. Only GFLB would need the MR. By that token perhaps the redeploy for voicing would be soon enough to handle this?

**Comment by terracoda** [comment]
Yes, I would like this to be fixed for the re-deploy of GFLB with Voicing.

As for GFL regular, the qualitative distance region parameters which include "farthest" are only used in the scene summary. This sim does not have an explicit qualitative distance exploration mode, the values for the distances between the spheres is described quantitatively in meters everywhere else in this sim.

The dynamic phrase for position containing the "farthest" region can be seen in the screenshot below. The complete wording for this phrase is different than the responsive descriptions in GFLB, and maybe .

So basically, GFL regular uses qualitative distance regions in its scene summary, but not in the mass spheres' dynamic state descriptions and not in the `aria-valuetext` for the sphere position sliders.

Screen shot of published version of GFL regular 
<img width="1431" alt="Screen Shot 2021-08-04 at 10 41 24 AM" src="https://user-images.githubusercontent.com/1902116/128187273-5821e256-d20f-46fd-985e-b10058e3d00f.png">

Here's a comparison screen shot of GFLB and all the places (underlined in red) where qualitative distance is used. In this screen shot the distance should be described as "farthest from", but it is described as "extremely far from".

**GFLB A11y View**
![Screen Shot 2021-08-04 at 10 53 26 AM](https://user-images.githubusercontent.com/1902116/128193737-944ebfea-7693-44ea-817c-a0013db322e5.png)

Oh, and here's the region in the object responses 
![gflb-response](https://user-images.githubusercontent.com/1902116/128194849-65345e7f-50d6-47e6-9330-2bdb43cf8e0d.png)

**Comment by jessegreenberg** [comment]
Change made in the above commits. I tested it in GFL and GFL:B and I am seeing"...farthest from..." in both now. @zepumph could you please review the code change, is this in line with what we discussed?

@terracoda would you like to review the behavior?

**Comment by zepumph** [comment]
Your implementation assumes that we will want to keep the exact same regions for Coulombs Law, which when we were designing and implementing things originally, I wasn't confident about. I'm happy to proceed if you and @terracoda decided that both GFL sims would be good using the same regions, and also as it is likely that CL will come along for free when its time comes. I like that the change meant that you could get rid of the duplication, but please note that the GFL and GFLB don't have the same distance regions. I thought the task here was to normalize each implementation of getDistanceIndex() for each sim's use.

For reference:

GFL distance region design: https://docs.google.com/document/d/1-37qAgde2XrlXBQae2SgjartM35_EnzDD9pdtd3nXAM/edit#heading=h.2ksj9snic7jq 
GFLB distance region design: https://docs.google.com/document/d/1HdDG9ds2MdbCb21l9qk3cI8yBQxK6-wBWNXC4Tloji8/edit#heading=h.5bw1acylrasd

**Comment by terracoda** [comment]
@zepumph, can you clarify what you mean by
> but please note that the GFL and GFLB don't have the same distance regions.  

Is it because they are not mathematically the same:
- GFL regular has a range of 1.1 meters (closest to) - 10 meters (farthest from)
- GFLB has a range from 1.3 km (closest to) to 10 km (farthest from)?

I'd like to clarify that it is the words used in the qualitative scale from "closest to" through "farthest from" that are the same in both sims, I a not sure if there are difference in the ranges for each region.

I have a feeling if we use percentages, the 2 sims can share the same scale, though I am not 100% sure. And I hope the percentages are robust enough to capture Coulomb's Law as well, but that sim has some differences like the presence of zero and negative values.

**Comment by jessegreenberg** [comment]
I had thought the intent behind the design in https://github.com/phetsims/gravity-force-lab-basics/issues/206#issuecomment-848960764 was that it would apply to all ISLC sims
> Will these percentage ranges work for all ISLC sims?

And that meant that the change in regions for both GFL and GFL:B were acceptable. I am hopeful this will work for CL as well when the time comes. If it doesn't we can either change this again or override the method of PositionDescriber for that sim.

**Comment by zepumph** [comment]
>And that meant that the change in regions for both GFL and GFL:B were acceptable. I am hopeful this will work for CL as well when the time comes. If it doesn't we can either change this again or override the method of PositionDescriber for that sim.


It seems like the path forward is for @terracoda to confirm this design. I didn't realize that was the goal at all. I thought we would normalize for each individual sim's needs, keeping the abstract method.

> I am not sure if there are differences in the ranges for each region.

@terracoda please take a look at the two tables I linked to in https://github.com/phetsims/gravity-force-lab-basics/issues/206#issuecomment-899737036. They show different ranges for the same regions. Please determine if they should be aligned, or kept the way they were designed. If they are being changed, please recommend a list of regions that are good for GFL and GFLB, and hopefully for CL as well when we get to it.

**Comment by zepumph** [comment]
>Is it because they are not mathematically the same:

@jessegreenberg I think the best path forward is to revert what was implemented, and to normalize each individual sim's implementation of the ranges.

**Comment by jessegreenberg** [comment]
Sounds good. I noticed that the GFL design doc was updated so I think this can be closed. Feel free to reopen if there is anything else for this issue.
